src/manager.ts

```diff
@@ -112,6 +112,89 @@ export class Manager implements vscode.TreeDataProvider<string | FileSystemConfi
         }
         root = root.replace(/^~/, home.replace(/\/$/, ''));
       }
+      
+      vscode.debug.registerDebugConfigurationProvider("*", {
+        resolveDebugConfiguration(folder: vscode.WorkspaceFolder, debugConfig: vscode.DebugConfiguration) {
+
+          if ((folder.uri.scheme === 'ssh') && config) {
+            debugConfig['request'] = 'attach';
+            debugConfig['pathMappings'] = [];
+            debugConfig['pathMappings'].push({ 'localRoot': 'ssh://' + config.name, 'remoteRoot': config.root });
+
+            if (config.debugPort) {
+              if (debugConfig['port'] && (debugConfig['port'] !== config.port))
+                Logging.warning(`Invalid port in 'launch.json' working on '${config.name}'`);
+              debugConfig['port'] = config.debugPort;
+
+              if (debugConfig['host'] && (debugConfig['host'] !== config.host))
+                Logging.warning(`Invalid host in 'launch.json' working on '${config.name}'`);
+              debugConfig['host'] = config.host;
+            }
+
+            if (debugConfig['port'] === undefined)
+              Logging.error(`Missing property "debugPort" for remote debugging on '${config.name}'`);
+
+            let workspaceFolder = root;
+            let file = "${file}"; // default to get error message to open an editor
+            if (vscode.window.activeTextEditor)
+              file = root + vscode.window.activeTextEditor.document.uri.path; // '${file}' can not be resolved per default              
+            if (debugConfig['program'])
+              debugConfig['program'] = eval('`' + debugConfig['program'] + '`'); // resolve ${file}, ${workspaceFolder}   
```

Comment:
This seems a bit unsafe and a bad practice. A regular (dynamic) string replacement would work here.
I'm also not sure about if vscode will replace things like `${workspaceFolder}` before/after this step.